sdk/spring/spring-cloud-azure-stream-binder-eventhubs/src/main/java/com/azure/spring/cloud/stream/binder/eventhubs/config/ClientFactoryCustomizer.java

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.cloud.stream.binder.eventhubs.config;
+
+import com.azure.spring.eventhubs.core.processor.EventHubsProcessorFactory;
+import com.azure.spring.eventhubs.core.producer.EventHubsProducerFactory;
+
+/**
+ * Called by the binder to customize the factories.
+ */
+public interface ClientFactoryCustomizer {
+
+    /**
+     * Customize the producer factory.
+     * @param factory The Event Hubs producer factory.
+     */
+    default void customize(EventHubsProducerFactory factory) {
```

Comment:
what about two customizer interfaces for different target 'factory'? Other places looks keep that of "xxxCustomizer" to have one method of "customize(xxx)", this makes it understandable. how do you feel?

Reply:
It's imitating the Kafka binder implementation.